src/components/Navigator/QuickNavigationModal.vue

```diff
@@ -9,180 +9,387 @@
 -->
 
 <template>
-  <div class="quick-navigation-modal">
-    <div class="container">
+  <div
+    class="quick-navigation"
+    @keydown.down.exact.prevent="focusNext"
+    @keydown.esc.exact.prevent="closeQuickNavigationModal"
+    @keydown.up.exact.prevent="focusPrev"
+    @keydown.enter.exact="handleKeyEnter"
+  >
+    <div
+      class="quick-navigation__modal-shadow"
+      @click.self="closeQuickNavigationModal"
+    >
       <div
-        class="close-icon"
-        @click="closeQuickNavigationModal()"
+        class="quick-navigation__container"
       >
-        <p>
-          Close
-        </p>
-      </div>
-      <input
-        ref="input"
-        class="filter"
-        v-model="userInput"
-        type="text"
-      />
-      <div>
+        <FilterInput
+          v-model="userInput"
+          placeholder="Search symbols"
+          class="quick-navigation__filter"
+          focusInputWhenCreated
+          focusInputWhenEmpty
+          @input="focusedIndex = 0"
+        >
+          <template slot="icon">
+            <div
+              class="quick-navigation__magnifier-icon-container"
+              :class="{ 'blue': userInput.length }"
+            >
+              <MagnifierIcon />
+            </div>
+          </template>
+          <template #input-menu-items>
+            <div
+              v-if="!userInput.length"
+              class="quick-navigation__open-key-container"
+            >
+              <kbd>Cmd+Shift+o</kbd>
```

Comment:
This text was a little confusing for me when I was trying this feature out since it wasn't clear that this a the keyboard shortcut for opening the modal since there was just the text on its own. You have to close the modal before the shortcut does anything, which is why it was a little confusing. It was also a little unclear to me why there were multiple shortcuts that could be used—this one and the `/` key. Also, the `cmd` modifier key might be specific to keyboards used in Apple platforms.

Minor detail. And TIL about the `<kbd>` element :P

Reply:
I agree that having the shortcut as part of the input text could be confusing, I'm going to remove it from the input text and try to find a better place for it, if any, in another iteration. Regarding the `cmd` key, AFAIK using `metaKey` works for both Mac and Windows. The idea behind having both shortcuts is that `/` is very common for opening similar tools on websites, and `cmd+shift+o` mimics Xcode's Open Quickly shortcut. Thanks for pointing this out, Marcus! It's been a bit tricky to figure out how to make the feature visible to the user.

Reply:
The `metaKey` aka `Win` key on Windows is almost never used in browser shortcuts, as its OS specific. Its more common to use `ctrl`, same for Linux I think.